### DISCUSSION TITLE
还有别的获取stoken v1的接口吗

### DISCUSSION THREAD

**Discussion by Yoimiya-993** [UIGF-org/mihoyo-api-collect#63] [Q&A]
getMultiTokenByLoginTicket不能用了之后，试了下这个接口：
[https://github.com/UIGF-org/mihoyo-api-collect/blob/main/hoyolab/user/token.md#%E9%80%9A%E8%BF%87game-token%E8%8E%B7%E5%8F%96stokenv1](https://github.com/UIGF-org/mihoyo-api-collect/blob/main/hoyolab/user/token.md#%E9%80%9A%E8%BF%87game-token%E8%8E%B7%E5%8F%96stokenv1)
发现返回的是stoken v2并不是文档里说的v1，还有别的返回stoken v1的接口吗

**Comment by TRIM5829** [comment]
> getMultiTokenByLoginTicket不能用了之后，试了下这个接口： https://github.com/UIGF-org/mihoyo-api-collect/blob/main/hoyolab/user/token.md#%E9%80%9A%E8%BF%87game-token%E8%8E%B7%E5%8F%96stokenv1 发现返回的是stoken v2并不是文档里说的v1，还有别的返回stoken v1的接口吗

这个接口我用了后提示错误-100，应该是请求头漏了某些参数。请问请求需要哪些参数才能获取到stoken？

**Comment by Yoimiya-993** [comment]
> > getMultiTokenByLoginTicket不能用了之后，试了下这个接口： https://github.com/UIGF-org/mihoyo-api-collect/blob/main/hoyolab/user/token.md#%E9%80%9A%E8%BF%87game-token%E8%8E%B7%E5%8F%96stokenv1 发现返回的是stoken v2并不是文档里说的v1，还有别的返回stoken v1的接口吗
> 
> 这个接口我用了后提示错误-100，应该是请求头漏了某些参数。请问请求需要哪些参数才能获取到stoken？

[getTokenByGameToken](https://github.com/UIGF-org/mihoyo-api-collect/blob/main/hoyolab/user/token.md#%E9%80%9A%E8%BF%87game-token%E8%8E%B7%E5%8F%96stokenv1)接口吗，`account_id`字段是数字你是不是传成字符串了？

**Comment by TRIM5829** [comment]
> > > getMultiTokenByLoginTicket不能用了之后，试了下这个接口： https://github.com/UIGF-org/mihoyo-api-collect/blob/main/hoyolab/user/token.md#%E9%80%9A%E8%BF%87game-token%E8%8E%B7%E5%8F%96stokenv1 发现返回的是stoken v2并不是文档里说的v1，还有别的返回stoken v1的接口吗
> > 
> > 
> > 这个接口我用了后提示错误-100，应该是请求头漏了某些参数。请问请求需要哪些参数才能获取到stoken？
> 
> [getTokenByGameToken](https://github.com/UIGF-org/mihoyo-api-collect/blob/main/hoyolab/user/token.md#%E9%80%9A%E8%BF%87game-token%E8%8E%B7%E5%8F%96stokenv1)接口吗，`account_id`字段是数字你是不是传成字符串了？

现在可以用了，并且成功获取到了stoken_v2，谢谢！（居然不是stoken_v1，有点神奇）
但是在接下来用stoken获取authkey时，尽管cookie中有stoken mid，请求头有ds，仍然会返回-10001。是又有哪里出错了吗

**Reply by zhoubugouduo** [reply]
为什么我获取不了啊,你能帮我看看吗
url = "https://api-takumi.mihoyo.com/account/ma-cn-session/app/getTokenByGameToken"
headers = {"x-rpc-app_id": "bll8iq97cem8", }
json_data = {"account_id": int(uid),"game_token": game_token}
response = requests.post(url=url, json=json_data, headers=headers)
print(response.text)
{"data":null,"message":"登录状态失效，请重新登录","retcode":-100}

**Comment by TRIM5829** [comment]
> > > > getMultiTokenByLoginTicket不能用了之后，试了下这个接口： https://github.com/UIGF-org/mihoyo-api-collect/blob/main/hoyolab/user/token.md#%E9%80%9A%E8%BF%87game-token%E8%8E%B7%E5%8F%96stokenv1 发现返回的是stoken v2并不是文档里说的v1，还有别的返回stoken v1的接口吗
> > > 
> > > 
> > > 这个接口我用了后提示错误-100，应该是请求头漏了某些参数。请问请求需要哪些参数才能获取到stoken？
> > 
> > 
> > [getTokenByGameToken](https://github.com/UIGF-org/mihoyo-api-collect/blob/main/hoyolab/user/token.md#%E9%80%9A%E8%BF%87game-token%E8%8E%B7%E5%8F%96stokenv1)接口吗，`account_id`字段是数字你是不是传成字符串了？
> 
> 现在可以用了，并且成功获取到了stoken_v2，谢谢！（居然不是stoken_v1，有点神奇） 但是在接下来用stoken获取authkey时，尽管cookie中有stoken mid，请求头有ds，仍然会返回-10001。是又有哪里出错了吗

明白了，是这里说的生成DS算法没用了。在米游社登录抓到的2.70.1的DS可以正常使用，并且返回的authkey可以获取到抽卡记录
难道说，旧版本的ds不能用了吗？

**Comment by BTMuli** [comment]
目前第三方应该都用的是 `stoken_v2`，个人哪怕获取到 `stoken_v1` 也要调用相关接口升级为 v2 的，~~所以这个 issue 其实有点没必要~~

接口请求具体代码，可以从以下几个来源获取（随便排的）：

- [本仓库](https://github.com/UIGF-org/mihoyo-api-collect)，目前处于消极维护状态
- [ApiFox](https://github.com/UIGF-org/mihoyo-api-collect/issues/39)，同样处于消极维护状态
- [UIGF 成员](https://github.com/orgs/UIGF-org/people) 的[相关仓库](https://uigf.org/zh/partnership.html)，这方面就比较杂，建议根据开发语言自行选择（具体的我也不清楚不过如果用 `ts` 的话可以参考[我的](https://github.com/BTMuli/TGAssistant)）

详细的请求参数什么的其实不太建议在 issue 中讨论，毕竟 [`discussion`](https://github.com/UIGF-org/mihoyo-api-collect/discussions) 就是干这个的
